### PR TITLE
[FEATURE] implements the suggest feature of elasticsearch

### PR DESCRIPTION
**Pull request by damobert**
It offers a simple term suggestion on all fields (termSuggestions(aTerm)) or the possibility to
use your own suggestion query (suggestion("my-suggestions", suggestionDefinition))

### CODE REVIEW

**Comment by damobert** [comment]
:+1:

File: README.md [hunk 1]
```diff
@@ -268,7 +268,7 @@ aggregation definition and a pre-configured `fieldBasedAggregation`. Both method
 You can nest aggregations by providing a parent name.
 
 * `aggregation($name, array $aggregationDefinition, $parentPath = NULL)` -- generic method to add a $aggregationDefinition under a path $parentPath with the name $name
-* `fieldBasedAggregation($name, $field, $type = "terms", $parentPath = NULL)` -- adds a simple filed based Aggregation of type $type with name $name under path $parentPath. Used for simple aggregations like sum, avg, min, max or terms
```
Comment (daniellienert) [inline README.md]: These changes may came in accidentally ;)

**Comment by daniellienert** [comment]
:+1:

File: README.md [hunk 2]
```diff
@@ -393,6 +393,50 @@ of `__typeAndSupertypes` containing `TYPO3.Neos:Document`.
...
+### Examples
+#### Add a simple suggestion to a query
+Simple suggestion that returns a suggestion based on the sent term
+
+```
+suggestions = $(Search.query(site)...termSuggestions('someTerm')}
+```
+You can access your suggestions inside your fluid template with 
+```
+{nodes.aggregations}
```
Comment (johannessteu) [inline README.md]: this should be {nodes.suggestions} or not?

**Comment by johannessteu** [comment]
Functionallity looks good by reading. Commented one line for the doc. There are also som ```"```that should be replaced with an single quote

File: Classes/Flowpack/ElasticSearch/ContentRepositoryAdaptor/Eel/ElasticSearchQueryBuilder.php [hunk 3]
```diff
@@ -502,6 +502,64 @@ protected function addSubAggregation($parentPath, $name, $aggregationConfigurati
...
+     * This method is used to create a simple term suggestion.
+     *
+     * Example Usage of a term suggestion
+     *
+     * nodes = ${Search....termSuggestions("aTerm")}
+     *
+     * Access all suggestions data with {nodes.suggestions} in your fluid template
+     *
+     * @param $name
+     * @param $text
```
Comment (aertmann) [inline Classes/Flowpack/ElasticSearch/ContentRepositoryAdaptor/Eel/ElasticSearchQueryBuilder.php]: missing types here

File: Classes/Flowpack/ElasticSearch/ContentRepositoryAdaptor/Eel/ElasticSearchQueryBuilder.php [hunk 4]
```diff
@@ -502,6 +502,64 @@ protected function addSubAggregation($parentPath, $name, $aggregationConfigurati
...
+     *
+     * Access all suggestions data with {nodes.suggestions} in your fluid template
+     *
+     * @param $name
+     * @param array $suggestionDefinition
+     * @return $this
+     */
+    public function suggestions($name, array $suggestionDefinition)
+    {
+        if (!array_key_exists("suggest", $this->request)) {
```
Comment (aertmann) [inline Classes/Flowpack/ElasticSearch/ContentRepositoryAdaptor/Eel/ElasticSearchQueryBuilder.php]: all other quotes are single

File: Classes/Flowpack/ElasticSearch/ContentRepositoryAdaptor/Eel/ElasticSearchQueryBuilder.php [hunk 5]
```diff
@@ -502,6 +502,64 @@ protected function addSubAggregation($parentPath, $name, $aggregationConfigurati
...
+     *   terms = TYPO3.TypoScript:RawArray {
+     *     field = "body"
+     *   }
+     * }
+     *
+     * nodes = ${Search....suggestion("my-suggestions", this.suggestionDefinition).execute()}
+     *
+     * Access all suggestions data with {nodes.suggestions} in your fluid template
+     *
+     * @param $name
```
Comment (aertmann) [inline Classes/Flowpack/ElasticSearch/ContentRepositoryAdaptor/Eel/ElasticSearchQueryBuilder.php]: missing type

File: Classes/Flowpack/ElasticSearch/ContentRepositoryAdaptor/Eel/ElasticSearchQueryBuilder.php [hunk 6]
```diff
@@ -502,6 +502,64 @@ protected function addSubAggregation($parentPath, $name, $aggregationConfigurati
...
+     * This method is used to create any kind of suggestion.
+     *
+     * Example Usage of a term suggestion for the fulltext search
+     *
+     * suggestionDefinition = TYPO3.TypoScript:RawArray {
+     *     text = "some text"
+     *   terms = TYPO3.TypoScript:RawArray {
+     *     field = "body"
+     *   }
+     * }
```
Comment (aertmann) [inline Classes/Flowpack/ElasticSearch/ContentRepositoryAdaptor/Eel/ElasticSearchQueryBuilder.php]: some indentation problems

**Comment by aertmann** [comment]
generally looks okay, a couple of minor comments.. also did you consider using [] instead of array() since the package requires 5.5+ anyway?

**Comment by damobert** [comment]
I fixed most of the mistakes. Where are those indentation problems exactly?

**Pull request merged**